### DISCUSSION TITLE
[JSON Response] Why numbers are encoded as strings?

### DISCUSSION THREAD

**Discussion by jenkin** [osm-search/Nominatim#3115] [Q&A] [answered]
JSON(V2) response format contains strings that represents numbers, ie. `lat`, `lon`, `boundingbox`. Geojson response uses numbers instead. Is there a reason for this behavior and this difference?

I think this is a bug, because JSON can represent numbers (integers and floats), so string is the wrong type to use here.

Maybe I missed something (ie. backward compatibility constraints), so let's discuss before opening an issue and a PR.

**Comment by lonvia** [comment, marked as answer]
Historical reasons (i.e. a bug that became a feature). It's not the only odd thing in the format. We are not going to change anything in the output because it would potentially break processing on the user side. There is going to be a v2 version of the API at some point that is a bit more sane.

**Reply by jenkin** [reply]
A `jsonv2` format is there, so maybe we can think about a `jsonv3` with the correct types. Waiting for the v2. But after years of _stringified numbers_, it may not be very useful. Thanks for the answer!

**Reply by lonvia** [reply]
We already have json formats without all the bugs. `geocodejson` is the preferred way to go nowadays.

**Reply by ChristopheLyonnaz** [reply]
Hello
It seems you recently updated the json output format to change lat  / lon from string to double values.
For instance, when using the example request : https://nominatim.openstreetmap.org/?addressdetails=1&q=bakery+in+berlin+wedding&format=json&limit=1, the output is no longer identical to what is given in example:

In documentation, output looks like (lat / long returned as strings):
```
{.
    "lat": "52.5460941",
    "lon": "13.35918",
}
```

whereas, since some days, the output is now (lat / long returned as double):
```
{
    "lat": 52.54274275,
    "lon": 13.36690305710228,
}
```

I couldn't find a release note describing this change. 
Can you please indicate me if this is a definitive output format ?
Moreover, as you wrote previously, does it mean json will become obsolete and it is recommended to migrate to geojson format ?
Regards,
Christophe.

**Reply by lonvia** [reply]
And there I thought I had reproduced the format bug for bug. This is a regression that will be fixed shortly.

**Reply by ChristopheLyonnaz** [reply]
Ok, thank you for this quick feedback. Do you want me to raise a issue in this project ?
Any idea when you plan to submit a fix ?

**Reply by lonvia** [reply]
Fix is already live. Should you find another regression, please open a new issue.

**Reply by ChristopheLyonnaz** [reply]
Great !! Many thanks ! That is efficiency ! 
Fix tested ok for me !
